File: bitmarket-server/src/test/java/com/bitwise/bitmarket/registry/CommandLineTest.java
package com.bitwise.bitmarket.registry;

import com.beust.jcommander.ParameterException;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class CommandLineTest {

    @Test
    public void shouldParseValidCommandLine() throws Exception {
        CommandLine cli = parseCli("--port", "1234");
        assertEquals(1234, cli.port);
    }

    @Test
    public void shouldHonorDefaultPort() throws Exception {
        CommandLine cli = parseCli();
        assertEquals(CommandLine.DEFAULT_PORT, cli.port);
    }

    @Test(expected = ParameterException.class)
    public void shouldRejectInvalidArguments() throws Exception {
        parseCli("--unknown-flag");
    }

    private CommandLine parseCli(String... arguments) {
        return CommandLine.fromArgList(arguments);
    }
}

File: coinffeine-server/src/test/scala/com/coinffeine/CommandLineTest.scala
package com.coinffeine

import com.beust.jcommander.ParameterException
import org.scalatest.FlatSpec
import org.scalatest.matchers.ShouldMatchers

class CommandLineTest extends FlatSpec with ShouldMatchers {

  "The command line parser" must "parse the port" in {
    parseCli("--port", "1234").port should be (1234)
  }

  it should "honor the default port" in {
    parseCli().port should be (CommandLine.DefaultPort)
  }

  it should "reject invalid arguments" in {
    evaluating(parseCli("--unknown-flag")) should produce [ParameterException]
  }

  private def parseCli(arguments: String*) = CommandLine.fromArgList(arguments)
}
